js/libs/build.js

```diff
@@ -1,344 +1,344 @@
-Fliplet.Widget.instance('interactive-map', function(widgetData) {
```

Comment:
The key thing was to remove this and get the widget data another way. This way is present on the Form Builder.

Reply:
@hcarneiro Can you remove the unnecessary changes from this PR and update the original description?

Reply:
@squallstar Can you take a look at this fix and see whether this is a legit fix or if the issue comes from how `Fliplet.Widget.instance()` works?

Reply:
@tonytlwu `Fliplet.Widget.instance()` registers a function and calls it straight away when it's registered:

https://github.com/Fliplet/fliplet-api/blob/41f8ccdcf0be79467a3281bc6316ea74a363ce3a/public/assets/fliplet-core/1.0/core.js#L1500

---

Likewise, when a widget is dropped into a page the code on fliplet-interact is calling the function again:

https://github.com/Fliplet/fliplet-api/blob/41f8ccdcf0be79467a3281bc6316ea74a363ce3a/public/assets/fliplet-interact/2.0/js/interact.js#L235-L237

The only thing I can think of is that the moment at which is called is too early? You might want to add some logs and verify whether your sync scripts are all loaded before the function is called.

If that's the case then we might have to change the code to wait for all scripts to execute like `Fliplet.runtime` does.

By the way I wonder why no other widgets in the system are affected by this issue that the interactive map is having.... Like we have far more complex widgets like LFD and Form builder which do this all the time.

Reply:
@squallstar The Form Builder uses `Fliplet.Widget.instance()` but the LFD doesn't.
However the Form Builder does force the iframe to refresh, and the Interactive Map doesn't.

Reply:
> By the way I wonder why no other widgets in the system are affected by this issue that the interactive map is having.... Like we have far more complex widgets like LFD and Form builder which do this all the time.

@squallstar My theory is that we're not sensitive enough to these issues. We get issues with widgets not rendering all the time.

And following @hcarneiro's comment, it does seem that it's best to resolve it correctly so that we can improve the reliability of widget live rendering -- especially now that we have a case that's easily replicable 👍 

---

So to clarify, is this now down to me to test because it's related to `fliplet-interact`? And a potential solution is to use something like:

```js
  Fliplet.require(arrayOfScripts).then(function () {
    if (isExisting || !hasNewAssets) {
      widgetIds.forEach(function (widgetId) {
        Fliplet.Widget.instance(widgetId);
      });
    }
  });
```